### PR TITLE
coverage: git fetch first & ignore cargo-test profraw files

### PR DESCRIPTION
**Pull request by def-**
As reported by Philip (via Slack)

### Motivation

  * This PR fixes a previously unreported bug.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: ci/plugins/mzcompose/hooks/post-command [hunk 1]
```diff
@@ -31,17 +31,22 @@ export_cov() {
       > coverage/"$BUILDKITE_JOB_ID"-"$(basename "$1")".lcov
 }
 
+find_profraw() {
+    # Ignore cargo-test files which are unrelated to us
+    find . -name '*.profraw' -and -not -path '*/llvm-cov-target/*' "$@"
```
Comment (philip-stoev) [inline ci/plugins/mzcompose/hooks/post-command:36]: How did those files end up in a CI job that was not running cargo-test? Were they a left-over from a previous CI job that happened to use the same machine? In this case, maybe we should do a better cleanup at job initialization?

Comment (def-) [reply ci/plugins/mzcompose/hooks/post-command:36]: Yes, it appears that processes from a previous cargo-test can still be running in the background. How can we collect them all and kill them? I hoped that the exiting cargo-test would kill all its child processes, but this seems not to be the case.

Comment (philip-stoev) [reply ci/plugins/mzcompose/hooks/post-command:36]: killall?

Comment (def-) [reply ci/plugins/mzcompose/hooks/post-command]: Done, run is green, hope it's fine now.

File: misc/python/materialize/cli/ci_coverage_pr_report.py [hunk 2]
```diff
@@ -39,12 +39,15 @@ def find_modified_lines() -> Coverage:
     base_branch = os.getenv("BUILDKITE_PULL_REQUEST_BASE_BRANCH", "main") or os.getenv(
         "BUILDKITE_PIPELINE_DEFAULT_BRANCH", "main"
     )
+    # Make sure we have the latest state to correctly identify the merge base
+    subprocess.run(["git", "fetch", "origin", base_branch], check=True)
```
Comment (nrainer-materialize) [inline misc/python/materialize/cli/ci_coverage_pr_report.py:43]: I was running into exactly the same when doing the Protobuf checks 😁

**Review by nrainer-materialize — APPROVED** [review]

**Pull request merged**